### PR TITLE
Improve EditBox, enable reorder between Nodes

### PR DESCRIPTION
**Pull request by zilongshanren**
fixes https://github.com/cocos-creator/fireball/issues/1933
fixes https://github.com/cocos-creator/fireball/issues/1932

### CODE REVIEW

**Comment by zilongshanren** [comment]
@pandamicro  @jareguo  求review

**Comment by zilongshanren** [comment]
测试例修改：https://github.com/cocos-creator/example-cases/pull/64

File: cocos2d/core/label/CCSGLabel.js [hunk 1]
```diff
@@ -328,7 +328,7 @@ _ccsg.Label = _ccsg.Node.extend({
 
         this._resetBMFont();
         //specify font family name directly
-        if (extName === null) {
+        if (extName === null || extName === '') {
```
Comment (jareguo) [inline cocos2d/core/label/CCSGLabel.js:331]: `!extName` 就行了

Comment (zilongshanren) [inline cocos2d/core/label/CCSGLabel.js:331]: ok

**Pull request closed without merging**